service/src/main/java/org/ehrbase/dao/access/jooq/ContributionAccess.java

```diff
@@ -180,21 +181,16 @@ public UUID commit(Timestamp transactionTime, UUID committerId, UUID systemId, C
         setState(Objects.requireNonNullElse(state, ContributionDef.ContributionState.COMPLETE));
 
         // audit attributes
-        if (committerId == null) {
-            //get current user from JVM
-            String defaultUser = System.getProperty("user.name");
-            //check for that user in the DB
-            String scheme = System.getProperty("host.name");
-            if (scheme == null)
-                scheme = "local";
-            committerId = new PersistedPartyProxy(this).getOrCreate(defaultUser, UUID.randomUUID().toString(), scheme, getServerConfig().getNodename(), "PARTY");
+        if (committerId != null) {
+            auditDetails.setCommitter(committerId);
```

Comment:
@jakesmolka shouldn't the committerId variable be a PartyProxy instead of a UUID? Not sure how this is modelled internally. For instance what happens if there is a committer name and identifiers? (if the comitter is PARTY_IDENTIFIED https://specifications.openehr.org/releases/UML/latest/index.html#Diagrams___18_1_83e026d_1433773264580_430294_7408)

Reply:
Thorough thinking again, thanks! In short: This works with all kinds of PartyProxy (which are implemented).

Longer: Formally, for this refactoring I only removed the creation of default values here, since that is happening centralized now.
But regarding Parties: Here in the access layer this audit's committer is directly corresponding to an entry in the party DB table, hence the UUID. The way the party table and its handling is implemented can handle all the types. Technically it has a `party_type` column, which allows distinct processing, which might include looking for identifiers in the `identifier` table.
For instance, the new default committer is actually a PARTY_IDENTIFIED (see here: https://github.com/ehrbase/ehrbase/pull/552/files/1e1db2070aab5c71f5f7bae1ab2ce995e9eb2c84#diff-005b654bd9e859813f3f0618563b53272587b48c2ed26e8fb2ffba98229a3871R79).